File: core/src/main/java/pcb/uwu/unit/derived/scalar/AngleUnit.java
package pcb.uwu.unit.derived.scalar;

import pcb.uwu.core.BaseUnit;
import pcb.uwu.core.UnitCounter;

import java.util.function.Function;

public class AngleUnit extends BaseUnit {

	protected AngleUnit() {
		super(0, "", "", "", Function.identity(), Function.identity());
	}

	@Override
	public Class<? extends BaseUnit> getBaseUnitType() {
		return null;
	}

	@Override
	public UnitCounter getUnitCounter() {
		return new UnitCounter();
	}
}

File: core/src/main/java/pcb/uwu/unit/scalar/AngleUnit.kt
package pcb.uwu.unit.scalar

import pcb.uwu.core.BaseUnit
import pcb.uwu.core.UnitCounter
import java.util.function.Function

open class AngleUnit protected constructor()
    : BaseUnit(0, "rad", "Radian", "Radians", Function.identity(), Function.identity())
{
    override val unitCounter: UnitCounter
        get() = UnitCounter()
    override val baseUnitType = AngleUnit::class.java
}